src/telemetry.ts

```diff
@@ -0,0 +1,183 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------*/
+
+import vscode = require('vscode');
+import TelemetryReporter from 'vscode-extension-telemetry';
+
+export const extensionId: string = 'ms-vscode.Go';
+const extensionVersion: string = vscode.extensions.getExtension(extensionId).packageJSON.version;
+const aiKey: string = 'AIF-d9b70cd4-b9f9-4d70-929b-a071c400b217';
+
+export function sendTelemetryEventForModulesUsage() {
```

Comment:
What is the motivation for having a separate function for every kind of telemetry event that needs to be logged instead of moving only the `sendTelemetryEvent` to this file and leave all the files other than the utilities untouched?

Reply:
My thinking was that it would be helpful to see all of the different telemetry events clearly in one file as opposed to having them throughout. It makes it really obvious what telemetry is being collected, and I liked the idea of being able to look in one file and quickly see all of the different metrics being collected. I also thought the GDPR comments made the code harder to read, so this might be a little simpler. But if you don't like this, I'm happy to just do the most minimal change.